tracer/build/_build/Build.Profiler.Steps.cs

```diff
@@ -38,9 +41,88 @@ partial class Build
+        .Executes(async () =>
+        {
+            var vcpkg = await GetVcpkg();
+
+            var vcpkgObjFolder = NativeBuildDirectory / "vcpkg";
+            var buildtreesRoot = vcpkgObjFolder / "buildtrees";
+            var packagesRoot = vcpkgObjFolder / "packages";
+            var downloadRoot = vcpkgObjFolder / "download";
+            vcpkg($"integrate install --x-buildtrees-root={buildtreesRoot} --x-packages-root={packagesRoot} --downloads-root={downloadRoot}");
+
+            async Task<Tool> GetVcpkg()
+            {
+                var vcpkgFilePath = string.Empty;
+
+                try
+                {
+                    vcpkgFilePath = ToolPathResolver.GetPathExecutable("vcpkg.exe");
+                }
+                catch (ArgumentException)
+                { }
+
+                if (File.Exists(vcpkgFilePath))
+                {
+                    return ToolResolver.GetLocalTool(vcpkgFilePath);
+                }
+
+                // Check if already downloaded
+                var vcpkgRoot = RootDirectory / "vcpkg";
+                var vcpkgExecPath = vcpkgRoot / "vcpkg.exe";
+
+                if (File.Exists(vcpkgExecPath))
+                {
+                    return ToolResolver.GetLocalTool($"{vcpkgExecPath}");
+                }
+
+                await DownloadAndExtractVcpkg(vcpkgRoot);
+                Cmd.Value(arguments: $"cmd /c {vcpkgRoot / "bootstrap-vcpkg.bat"}");
+                return ToolResolver.GetLocalTool($"{vcpkgRoot / "vcpkg.exe"}");
+            }
+
+            async Task DownloadAndExtractVcpkg(string destinationFolder)
+            {
+                var nbTries = 0;
+                var keepTrying = true;
+                var vcpkgZip = TempDirectory / "vcpkg.zip";
+                using var client = new HttpClient();
+                const string vcpkgVersion = "2024.11.16";
+                while (keepTrying)
+                {
+                    nbTries++;
+                    try
+                    {
+                        var response = await client.GetAsync($"https://github.com/microsoft/vcpkg/archive/refs/tags/{vcpkgVersion}.zip");
+                        response.EnsureSuccessStatusCode();
+                        await using var stream = await response.Content.ReadAsStreamAsync();
+                        await using var file = File.Create(vcpkgZip);
+                        await stream.CopyToAsync(file);
+                        keepTrying = false;
+                    }
+                    catch (HttpRequestException)
+                    {
+                        if (nbTries > 3)
+                        {
+                            throw;
+                        }
+                    }
+                }
+
+                var tempFolder = TempDirectory / "vcpkg_temp";
+                CompressionTasks.UncompressZip(vcpkgZip, tempFolder);
+
+                CopyDirectoryRecursively(tempFolder / $"vcpkg-{vcpkgVersion}", destinationFolder);
```

Comment:
could not move the directory? (not enough permissions 🤷 )

Reply:
Weird 🤔 Meh, not a big deal (unless it's a particularly big file)

Reply:
I can retry if you want and we can investigate this  (currently looking at the size of the extracted package)